tools/OutlineService/OutlineService/OutlineService.cs

```diff
@@ -311,29 +301,68 @@ public partial class OutlineService : ServiceBase {
     public void ResetRouting(string proxyIp, string proxyInterfaceName) {
       // Proxy server.
       if (proxyIp != null) {
-        try {
-          DeleteProxyRoute(proxyIp, proxyInterfaceName);
-        } catch (Exception e) {
-          eventLog.WriteEntry($"failed to remove route to proxy server: {e.Message}", EventLogEntryType.Error);
-        }
+        DeleteProxyRoute(proxyIp, proxyInterfaceName);
       } else {
         eventLog.WriteEntry("cannot remove route to proxy server, have not previously set",
             EventLogEntryType.Warning);
       }
 
       this.proxyIp = null;
       this.gatewayInterfaceName = null;
+      this.gatewayIp = null;

```

Comment:
How about calling your new function `SetGatewayProperties`?

Reply:
Done, updated it to take null values.